### ISSUE TITLE
Depend on stevadore for the docker stuff

### ISSUE THREAD

**Issue by MarkEdmondson1234** [cloudyr/googleComputeEngineR#124]
This looks like the successor to harbor

https://github.com/richfitz/stevedore

**Comment by hongooi73** [comment]
Something to consider might be, rather than wrestling with the API, just call the commandline tool directly. This is what I do with [AzureContainers](https://github.com/cloudyr/AzureContainers) which has `call_docker`, `call_kubectl` and `call_helm` functions.

Remember that if the user has the Docker API available, that means they will also have the docker[.exe] binary. It's kind of hackish, but it gets the job done and works everywhere the binary is available, including Windows.

The only hassle might be if you want to read the output of the binary back into your script, but so far, I haven't had to deal with that.

The official Azure CLI does this trick as well; when you run `az aks` to interface with a Kubernetes cluster, you're actually calling the kubectl executable under the hood.

**Comment by MarkEdmondson1234** [comment]
Thats actually how the current setup works, derived from https://github.com/wch/harbor - but bear in mind its not local docker but docker upon the instance.  Its basically SSH commands to the instance that trigger `docker X`.  Its ok, but was a bit of pain making sure the SSH connection works on all systems (use `system2` rather than `system`) - and for instance on Windows a ssh client isn't present by default.  To get around that I suggested using the SSH client bundled with RStudio.exe

**Comment by hongooi73** [comment]
Hmm, how would you access the Docker API remotely? If you can run R on the instance (which you need to be able to use the stevedore package) then you can run the docker executable. Or do you intend to open a port on the VM to communicate with the Docker API from the local machine?

**Comment by MarkEdmondson1234** [comment]
Yes it goes via SSH to the VM, then executes docker on there:

https://github.com/cloudyr/googleComputeEngineR/blob/83b5dc2a9b0dba1dc977d5b1158ee083b45ad802/R/harbor_gce.R#L16-L26

But now I understand your point, and I agree I don't think stevadore can replace that, was thinking perhaps it may help with the Docker files or configurations perhaps.

**Comment by hongooi73** [comment]
Totally unrelated to stevedore now, but I just found out that there is an ssh package on CRAN: https://github.com/ropensci/ssh

**Comment by MarkEdmondson1234** [comment]
Yes I have an issue for it :) https://github.com/cloudyr/googleComputeEngineR/issues/92 - anything done by Jerom is super stable so hopefully will get rid of some flakey system calls by migrating over
